src/Fields/Fields.jl

```diff
@@ -72,6 +75,10 @@ Base.propertynames(field::Field) = propertynames(getfield(field, :values))
 @inline Base.getproperty(field::Field, name::Integer) =
     Field(getproperty(field_values(field), name), axes(field))
 
+Base.@propagate_inbounds function Base.getindex(field::ColumnField, i::Integer)
+    Base.getindex(field_values(field), i)
```

Comment:
If the goal is to eventually use `PlusHalf` types for typesafe indexing (noted here: https://github.com/CliMA/TurbulenceConvection.jl/issues/256#issuecomment-917803482) then you need to use the FiniteDifference `getidx` methods (`getidx(field_data, ::Interior, idx)`) operator instead of punching through to the datalayout getindex call.

Reply:
Yeah, that's the eventual goal, although I think we're still a bit aways away. I'm fine with making this change if it works either way

Reply:
Hmm, `Interior` (and `getidx`) is defined in the `Operators` module, which is included after the `Field` module. Should we define a `getidx` function stub and move `Interior` def into the `Field` module?

Reply:
Hmm, maybe we want to punt on this for now and discuss what exactly we need on the API side to support the changes you need.

Reply:
Sounds good

Reply:
I would like to use `PlusHalf` for indexing faces: our experience so far is that it catches a _lot_ of off-by-one errors. Why don't we move the `PlusHalf` into the `Fields` module, and then define `Base.getindex`.

Reply:
We can leave `getidx` and `Interior` in `Operators` for now. We don't need those for this.

Reply:
Ok, so are you proposing we only support `Base.getindex(field::FaceFiniteDifferenceField, ::PlusHalf)` and `Base.getindex(field::CenterFiniteDifferenceField, ::Int)`?

Reply:
Yes, exactly

Reply:
Hmm, I'm not sure how we can do that, since [TC.jl's 255](https://github.com/CliMA/TurbulenceConvection.jl/pull/255) only replaces the grid to start with. The indices that we traverse ([here](https://github.com/CliMA/TurbulenceConvection.jl/blob/main/src/Grid.jl#L59)) are used both for the coordinates (which would be fine for our grid) and for field variables that aren't using ClimaCore.Field's yet (which would be problematic).

Reply:
I also don't mind pirating these methods if that's preferred.

Reply:
I would go with pirating what you need for now.

Reply:
Maybe I'll just do that for now. This is kind of holding up my primary task.